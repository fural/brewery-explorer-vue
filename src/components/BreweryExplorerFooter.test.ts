import { fireEvent, render } from "@testing-library/vue";
import component from "./BreweryExplorerFooter.vue";

test("it should match snapshot", async () => {
  // Arrange
  const { container, getByText, getByTestId } = render(component, {
    props: { total: 100 },
  });

  const backToTop = getByTestId("backToTopButton");

  // Assert
  expect(getByText(/Number of results: 100/i)).toBeTruthy();
  expect(backToTop).toBeTruthy();
  expect(getByText("Â© 2023 Brewery Explorer Ltd")).toBeTruthy();
  expect(container).toMatchSnapshot();
});

test("it should emit the back to top event", async () => {
  // Arrange
  const { getByTestId, emitted } = render(component, {
    props: { total: 100 },
  });

  const backToTop = getByTestId("backToTopButton");

  // Act
  await fireEvent.click(backToTop);

  // Assert
  expect(emitted("backToTop")).toBeTruthy();
});
