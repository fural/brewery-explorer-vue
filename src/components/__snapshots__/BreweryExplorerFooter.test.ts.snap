// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it should match snapshot 1`] = `
<div>
  <div
    class="flex bg-blue-900 sticky bottom-0 h-16 mr-2 text-white text-sm"
  >
    <div
      class="flex flex-1 items-center p-4"
    >
      <div
        class="flex w-full justify-between text-center"
      >
        <div
          data-testid="numberOfResults"
        >
          Number of results: 100
        </div>
        <div>
          Â© 2023 Brewery Explorer Ltd
        </div>
      </div>
    </div>
    <div
      class="flex items-center justify-center w-20"
    >
      <button
        class="flex items-center justify-center bg-blue-800 h-full w-full"
        data-testid="backToTopButton"
      >
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.5 10.5L12 3m0 0l7.5 7.5M12 3v18"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
